src/ledger/LedgerManagerImpl.cpp

```diff
@@ -947,6 +951,21 @@ LedgerManagerImpl::closeLedger(LedgerCloseData const& ledgerData)
     // step 4
     mApp.getBucketManager().forgetUnreferencedBuckets();
 
+    // Spin for parametrized amount of time in simulation mode
+    if (mApp.getMode() == Application::AppMode::RUN_CONSENSUS_ONLY)
+    {
+        auto spinFor = std::chrono::milliseconds(
+            mApp.getConfig().ARTIFICIALLY_SIMULATE_TX_APPLY_MS_FOR_TESTING);
+        auto applicationTime = closeLedgerTime.checkElapsedTime();
+        if (applicationTime < spinFor)
+        {
+            spinFor -= applicationTime;
```

Comment:
I don't think we need this extra complexity here: always add the delay as we simulate I/O overhead (so whatever time was spent processing the tx set so far is "real").
We can always do something like this in the future if we have to (based on how the metrics look like)

Reply:
What you suggested was my original approach. I did not like the deviation in metrics that I saw though: in a small sample of ledgers under high load, "real" application time went up to 260 ms. Since I spin for 2 seconds, my simulation was actually off by 13%, which seemed too high, so I wanted to be a bit more precise. What do you think?